src/sage/topology/simplicial_complex.py

```diff
@@ -4802,7 +4802,95 @@ def intersection(self, other):
             F = F + [s for s in self.faces()[k] if s in other.faces()[k]]
         return SimplicialComplex(F)
 
+    #@cached_method    tried this, but does not make much sense to cache this
+    def bigraded_betti_numbers(self):
+        r"""
+        Returns a dictionary of the bigraded Betti numbers of ''self'', 
+        with keys '(-i, 2j)'.
+
+        See bigraded_betti_number(self, a, b) for more information.
+
+        EXAMPLES::
 
+            sage: X = SimplicialComplex([[0,1],[1,2],[1,3],[2,3]])
+            sage: Y = SimplicialComplex([[1,2,3],[1,2,4],[3,5],[4,5]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 2, (-1, 6): 1, (-2, 6): 1, (-2, 8): 1}
+            sage: Y.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 3, (-2, 6): 1, (-2, 8): 2, (-3, 10): 1}
+        """
+        from sage.misc.functional import gens 
+        L = set(self.vertices())
+        n = len(L)
+        D = {}
+        B = {}
+        Point = SimplicialComplex([[0]])
+        H0 = Point.homology().get(0)
+
+        for i in range(n+1):
+            D[i] = list(combinations(L,i))
+
+        for i in range(n+1):
+            for j in range (n+1):
+                B[(-i, 2*j)] = self.bigraded_betti_number(-i, 2*j) 
+
+        #B[(0,0)] = 1
+
+        #for j in range(n+1):
+            #for x in D[j]:
+                #S = self.generated_subcomplex(x)
+                #H = S.homology()
+                #for k in range(j):
+                    #if H.get(j-k-1) != H0 and H.get(j-k-1) != None:
+                        #B[(-k, 2*j)] = B[(-k, 2*j)] + len(gens(H.get(j-k-1)))
+
+        B = {key:val for key,val in B.items() if val != 0}
+	    
+        return B
+
+    @cached_method
+    def bigraded_betti_number(self, a, b):
+        r"""
+        Returns the bigraded Betti number indexed in the form (-i, 2j).
+
+        Bigraded Betti number with indices '(-i, 2j)' is defined as a sum of ranks
+        of '(i-j-1)'-th (co)homologies of full subcomplexes with exactly 'j' vertices. 
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[2,0],[1,2,3]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, ,(-1,4): 1, (-1, 6): 1, (-2, 8): 1}
+            sage: X.bigraded_betti_number(-1, 6)
+            1
+        """
+        from sage.misc.functional import gens
+        if b % 2 == 1:
+            return 'Error: Second argument must be even.'
+        if a == 0 and b == 0:
+            return 1;
+
+        a = -a
+        b = b / 2
+        L = set(self.vertices())
+        n = len(L)
+        D = {}
+        Point = SimplicialComplex([[0]])
+        H0 = Point.homology().get(0)
+
+        for i in range(n + 1):
+            D[i] = list(combinations(L, i))
+
+        B = 0
+
+        for x in D[b]:
```

Comment:
You don't need to build `D`:
```suggestion
        for x in combinations(L, b):
```